socorro/external/es/crashstorage.py

```diff
@@ -555,45 +559,59 @@ def _submit_crash_to_elasticsearch(
         for _ in range(5):
             try:
                 with self.client() as conn:
-                    return self._index_crash(
-                        conn, index_name, es_doctype, crash_document, crash_id
-                    )
+                    return self._index_crash(conn, index_name, crash_document, crash_id)
 
-            except elasticsearch.exceptions.ConnectionError:
+            except elasticsearch.ConnectionError:
                 # If this is a connection error, sleep a second and then try again
                 time.sleep(1.0)
 
-            except elasticsearch.exceptions.TransportError as e:
-                # If this is a TransportError, we try to figure out what the error
+            except elasticsearch.BadRequestError as e:
+                # If this is a BadRequestError, we try to figure out what the error
                 # is and fix the document and try again
```

Comment:
this section removes fields that cause `document_parsing_exception` and retries the document. This seems like an odd choice given that it happens after value fixing occurs, which should already be preventing the three types of failure we catch here. The only way i can think of to reach this code block in production is if we are writing to a field not in our mapping.

Reply:
It used to be the case that it wrote _all_ the data into Elasticsearch even if it wasn't in the mapping. That way when they add new fields to the crash report, they'd get indexed even if Socorro didn't explicitly have support for it. While the intentions were good, that was _terrible_ so I changed it such that it _only_ indexes what's defined in super search fields and in the mapping.